src/parquet/statistics.cc

```diff
@@ -96,6 +96,67 @@ void TypedRowGroupStatistics<DType>::Reset() {
   has_min_max_ = false;
 }
 
+template <typename T>
+inline int getValueBeginOffset(const T* values, int64_t count) {
+  return 0;
+}
+
+template <typename T>
+inline int getValueEndOffset(const T* values, int64_t count) {
+  return count;
+}
+
+template <typename T>
+inline bool notNaN (const T* value) {
+  return true;
+}
+
+template <>
+inline int getValueBeginOffset<float>(const float* values, int64_t count) {
```

Comment:
If we used `std::is_floating_point` and the functor pattern, we could avoid code duplication

```c++
template <typename T, typename Enable = void>
struct StatsHelper {
  ...
};

template <typename T>
struct StatsHelper<T, typename std::enable_if<std::is_floating_point<T>::value>::type> {
  ...
};
```

Reply:
Thanks for this tip. Very helpful! Will include other review comments.